metadata-ingestion/tests/unit/stateful_ingestion/provider/test_file_ingestion_checkpointing_provider.py

```diff
@@ -0,0 +1,86 @@
+from typing import List
+
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.ingestion_job_checkpointing_provider_base import JobId
+from datahub.ingestion.source.state.checkpoint import Checkpoint
+from datahub.ingestion.source.state.sql_common_state import (
+    BaseSQLAlchemyCheckpointState,
+)
+from datahub.ingestion.source.state.usage_common_state import (
+    BaseTimeWindowCheckpointState,
+)
+from datahub.ingestion.source.state_provider.file_ingestion_checkpointing_provider import (
+    FileIngestionCheckpointingProvider,
+)
+from tests.test_helpers.type_helpers import assert_not_null
+
+pipeline_name: str = "test_pipeline"
+job_names: List[JobId] = [JobId("job1"), JobId("job2")]
+run_id: str = "test_run"
+
+
+def test_file_ingestion_checkpointing_provider(tmp_path):
```

Comment:
this code looks awfully similar to the code in https://github.com/datahub-project/datahub/blob/cc94ffbf6cb4c17cceb1f105eb1da6217bc5ddd3/metadata-ingestion/tests/unit/stateful_ingestion/provider/test_datahub_ingestion_checkpointing_provider.py#L27 - can you refactor to reduce the amount of duplicated code?